src/calculateDeepEqualDiffs.js

```diff
@@ -46,6 +46,14 @@ function accumulateDeepEqualDiffs(a, b, diffsAccumulator, pathString = ''){
       trackDiff(a, b, diffsAccumulator, pathString, diffTypes.different)
   }
 
+  if(isSet(a) && isSet(b)){
+    if(a.size !== b.size){return trackDiff(new Set(a), new Set(b), diffsAccumulator, pathString, diffTypes.different)}
```

Comment:
there's no need to re-create the sets here and under when reporting results

Reply:
The goal was to freeze the original sets to defend against situations where the originals may have since changed (and then the console may report on mutated data).

Reply:
i thought about it for some time. i think cloning the objects should indeed happen for all the different objects in this file

Reply:
One tradeoff (including the cloned Sets I added) is that I assume we'll be consuming much more memory? I wonder if the garbage collector will be able to reap the objects once we've logged them and moved on.